Quartz/Engine/Include/Quartz/Core/Graphics/API/GL/GLRenderDevice.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Genten Studios
+// 
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the 
+// following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the 
+// following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the 
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote 
+// products derived from this software without specific prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED 
+// WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A 
+// PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY 
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, 
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH 
+// DAMAGE.
+
+#pragma once
+
+#include <Quartz/Core/Graphics/API/IRenderDevice.hpp>
+#include <Quartz/Core/Utilities/HandleAllocator.hpp>
+#include <Quartz/Core/Graphics/API/InputLayout.hpp>
+#include <Quartz/Core/Graphics/API/GL/GLVertexBuffer.hpp>
+#include <Quartz/Core/Graphics/API/GL/GLShaderPipeline.hpp>
+#include <Quartz/Core/Graphics/API/GL/GLTexture.hpp>
+
+namespace qz { namespace gfx { namespace api { namespace gl {
```

Comment:
Namespace declarations on new lines, might waste space, but much more dev friendly when we come back to read it once we've forgotten everything about it.